src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -30,6 +31,11 @@ internal sealed partial class SolutionCompilationState
     /// </summary>
     private static readonly ConditionalWeakTable<ISymbol, ProjectId> s_assemblyOrModuleSymbolToProjectMap = new();
 
+    /// <summary>
+    /// Reusable empty dictionary
+    /// </summary>
+    private static readonly ReadOnlyDictionary<ProjectId, ICompilationTracker> s_emptyReadOnlyDictionary = new(new Dictionary<ProjectId, ICompilationTracker>());
```

Comment:
💡 If we make a change here, it should be a change to using `ImmutableSegmentedDictionary<ProjectId, ICompilationTracker>`.

Reply:
I can create a PR with changes to that and run speedometer too. It's not obvious to me that ImmutableSegmentedDictionary is the right choice here, but let me give it a go and we'll see what the numbers say.